modal/cls.py

```diff
@@ -160,6 +169,11 @@ def _get_metadata(self) -> api_pb2.ClassHandleMetadata:
             )
         return class_handle_metadata
 
+    def with_options(self, **kwargs: Any) -> "_Cls":
+        self._options = _parse_function_options(**kwargs)
```

Comment:
you don't want to mutate the object here right? you want to return a new object with this set

Reply:
i'm worried if I have the wrong approach

```python

    
def _initialize_from_other(self, other: "_Cls"):
    self._user_cls = other._user_cls
    self._functions = other._functions
    self._options = other._options
    self._callables = other._callables
    self._output_mgr: Optional[OutputManager] = other._output_mgr
    
 # and something like this
 
 
@classmethod
def from_existing(existing_cls):
    """Produces a new Cls from an existing Cls"""

    cls = _Cls._from_loader(
        existing_cls,
        existing_cls._load,
        existing_cls._rep,
        existing_cls._is_another_app,
        existing_cls._preload,
        existing_cls._hydrate_lazily,
        existing_cls._deps,
    )
    cls._initialize_from_existing(existing_cls)
        
 
 

```

Reply:
not following – i'm saying `with_options` should not mutate `self` but return a new object instead